doc/source_list_library.tex

```diff
@@ -45,26 +47,26 @@ \section*{Appendix: List library}
 
 function build_list(n, fun){
     function build(i, fun, already_built) {
-	return i < 0
-	    ? already_built
-	    : build(i - 1, fun, pair(fun(i),
-		  		     already_built));
+        return i < 0
+            ? already_built
+            : build(i - 1, fun, pair(fun(i),
+                already_built));
     }
-    return build(n - 1, fun, []);
+    return build(n - 1, fun, null);
 }
 
 // for_each applies first arg fun, assumed to be a unary function,
 // to the elements of the second argument, assumed to be a list.
 // fun is applied element-by-element:
-// for_each(fun, [1, [2, []]]) results in the calls fun(1) and fun(2).
+// for_each(fun, [1, [2, null]]) results in the calls fun(1) and fun(2).
```

Comment:
Same as `map` above, might want to replace literal list syntax.

Reply:
noted!